File: shopping_mall/frontend/src/admin/types/expense.ts
export interface ExpenseEntry {
  id: number;
  date: string;
  description: string;
  amount: number;
  category: string | null;
  auto_classified: boolean;
  created_at: string;
}

export interface ExpenseCreatePayload {
  date: string;
  description: string;
  amount: number;
  category?: string;
}

File: shopping_mall/backend/app/models/expense.py
from datetime import datetime
from sqlalchemy import Boolean, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base


class ExpenseEntry(Base):
    __tablename__ = "expense_entries"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    date: Mapped[str] = mapped_column(String(10), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)
    amount: Mapped[int] = mapped_column(Integer, nullable=False)
    category: Mapped[str | None] = mapped_column(String(30), nullable=True)
    auto_classified: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[datetime] = mapped_column(default=func.now())
